hack/ci/e2e-k8s.sh

```diff
@@ -22,167 +22,9 @@ set -o errexit -o nounset -o xtrace
-  # c.k8s-prow-builds.internal google.internal.
-  # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
-  # otherwise pods stops trying to resolve the domain.
-  if [ "${IP_FAMILY:-ipv4}" = "ipv6" ]; then
-    # Get the current config
-    original_coredns=$(kubectl get -oyaml -n=kube-system configmap/coredns)
-    echo "Original CoreDNS config:"
-    echo "${original_coredns}"
-    # Patch it
-    fixed_coredns=$(
-      printf '%s' "${original_coredns}" | sed \
-        -e 's/^.*kubernetes cluster\.local/& internal/' \
-        -e '/^.*upstream$/d' \
-        -e '/^.*fallthrough.*$/d' \
-        -e '/^.*forward . \/etc\/resolv.conf$/d' \
-        -e '/^.*loop$/d' \
-    )
-    echo "Patched CoreDNS config:"
-    echo "${fixed_coredns}"
-    printf '%s' "${fixed_coredns}" | kubectl apply -f -
-  fi
-
-  # ginkgo regexes
-  SKIP="${SKIP:-}"
-  FOCUS="${FOCUS:-"\\[Conformance\\]"}"
-  # if we set PARALLEL=true, skip serial tests set --ginkgo-parallel
-  if [ "${PARALLEL:-false}" = "true" ]; then
-    export GINKGO_PARALLEL=y
-    if [ -z "${SKIP}" ]; then
-      SKIP="\\[Serial\\]"
-    else
-      SKIP="\\[Serial\\]|${SKIP}"
-    fi
-  fi
-
-  # setting this env prevents ginkgo e2e from trying to run provider setup
-  export KUBERNETES_CONFORMANCE_TEST='y'
-  # setting these is required to make RuntimeClass tests work ... :/
-  export KUBE_CONTAINER_RUNTIME=remote
-  export KUBE_CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
-  export KUBE_CONTAINER_RUNTIME_NAME=containerd
-  ./hack/ginkgo-e2e.sh \
-    '--provider=skeleton' "--num-nodes=${NUM_NODES}" \
-    "--ginkgo.focus=${FOCUS}" "--ginkgo.skip=${SKIP}" \
-    "--report-dir=${ARTIFACTS}" '--disable-log-dump=true'
-}
-
-main() {
-  # create temp dir and setup cleanup
-  TMP_DIR=$(mktemp -d)
-  trap cleanup EXIT
-
-  # ensure artifacts (results) directory exists when not in CI
-  export ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"
-  mkdir -p "${ARTIFACTS}"
-
-  # export the KUBECONFIG to a unique path for testing
-  KUBECONFIG="${HOME}/.kube/kind-test-config"
-  export KUBECONFIG
-  echo "exported KUBECONFIG=${KUBECONFIG}"
-
-  # debug kind version
-  kind version
-
-  # default to bazel
-  # TODO(bentheelder): remove this line once we've updated CI to explicitly choose
-  BUILD_TYPE="${BUILD_TYPE:-bazel}"
-
-  # build kubernetes
-  if [ "${BUILD_TYPE:-}" = "bazel" ]; then
-    build_with_bazel
-  else
-    build
-  fi
-
-  # in CI attempt to release some memory after building
-  if [ -n "${KUBETEST_IN_DOCKER:-}" ]; then
-    sync || true
-    echo 1 > /proc/sys/vm/drop_caches || true
-  fi
-
-  # create the cluster and run tests
-  create_cluster && run_tests
-}
-
-main
+# BUILD_TYPE: bazel or docker
+#
+export KIND_BUILD_TYPE="${BUILD_TYPE:-bazel}"
+export KIND_TEST_FOCUS_OVERRIDE="${FOCUS:-}"
+export KIND_TEST_SKIP_OVERRIDE="${SKIP:-}"
+kind test presubmit --ip-family="${IP_FAMILY:-ipv4}"
```

Comment:
intend to minimize knobs that need touching heavily to avoid the typical confusion we have with our test rigs, however ip-family is one axis that we definitely test along in kubernetes presubmit and can't really handle well otherwise.

when we migrate this into prow config it will be simpler like:
`kind test presubmit`
`kind test presubmit --ip-family=ipv6`
w/o the env stuff.

Reply:
just that you mention this topic, want to share something I realized this week.
With dual-stack is it possible to have 2 kind of clusters:
- dualstack ipv4 (default) - ipv6 (sec)
- dualstack ipv6 (default) - ipv4 (sec)

maybe something we should think for the future, the default is giving by the order of the cluster-cidr configured.

Reply:
that isn't really related to this change though.

Reply:
why I can't resolve the review comments to minimize them?
🤔

Reply:
you don't have permissions. :github: